astropy/units/quantity.py

```diff
@@ -314,6 +315,96 @@ class Quantity(np.ndarray):
 
     __array_priority__ = 10000
 
+    def __class_getitem__(cls, unit_shape_dtype):
+        """Quantity Type Hints.
+
+        Unit-aware type hints are ``Annotated`` objects that encode the class,
+        the unit, and possibly shape and dtype information, depending on the
+        python and :mod:`numpy` versions.
+
+        Schematically, ``Annotated[cls[shape, dtype], unit]``
+
+        As a classmethod, the type is the class, ie ``Quantity``
+        produces an ``Annotated[Quantity, ...]`` while a subclass
+        like :class:`~astropy.coordinates.Angle` returns
+        ``Annotated[Angle, ...]``.
+
+        Parameters
+        ----------
+        unit_shape_dtype : :class:`~astropy.units.UnitBase`, str, `~astropy.units.PhysicalType`, or tuple
+            Unit specification, can be the physical type (ie str or class).
+            If tuple, then the first element is the unit specification
+            and all other elements are for `numpy.ndarray` type annotations.
+            Whether they are included depends on the python and :mod:`numpy`
+            versions.
+
+        Returns
+        -------
+        `typing.Annotated`, `typing_extensions.Annotated`, `astropy.units.Unit`, or `astropy.units.PhysicalType`
+            Return type in this preference order:
+            `typing.Annotated` if python v3.9+; `typing_extensions.Annotated`
+            if :mod:`typing_extensions` is installed, a `astropy.units.Unit`
```

Comment:
Maybe simplify: "... is installed; or an `...`" and then delete "else-wise, depending on the input."

Reply:
I just made it a bullet point list.